haystack/preview/document_stores/errors.py

```diff
@@ -2,6 +2,10 @@ class StoreError(Exception):
     pass
 
 
+class FilterError(StoreError):
```

Comment:
let's make this consistent, the in-memory document store used to raise both a specific `XXXFilterError` and `ValueError`